src/java/com/reucon/openfire/plugin/archive/model/ArchivedMessage.java

```diff
@@ -77,9 +77,15 @@ public ArchivedMessage(@Nullable final Long id, @Nonnull final Date time, @Nonnu
         this.with = with;
         this.body = body;
 
-        if ( stanza != null) {
-            final Document doc = DocumentHelper.parseText( stanza );
-            this.stanza = new Message( doc.getRootElement() );
+        if ( stanza != null && stanza.length() > 0 ) {
+            Message stanzaResult;
+            try {
+                final Document doc = DocumentHelper.parseText( stanza );
+                stanzaResult = new Message( doc.getRootElement() );
+            } catch (DocumentException de) {
+                stanzaResult = null;
```

Comment:
```suggestion
                Log.debug("Unable to parse (non-empty) stanza (id: {})", id, de);
                stanzaResult = null;
```
I've spent countless hours chasing down issues where my life would've been easier if some kind of exception would not have been ignored, but logged instead. Please always log an exception somewhere, if only to debug.

When the stanza was saved to the database (as it's non-empty), and parsing still fails, something is off, that even more so warrants some kind of log line.

Reply:
Absolutely right. I've got a bunch of branches with extra debug logging.